cpp/src/arrow/compute/kernels/vector_hash.cc

```diff
@@ -697,6 +698,12 @@ void AddHashKernels(VectorFunction* func, VectorKernel base, OutputType out_ty)
     base.signature = KernelSignature::Make({ty}, out_ty);
     DCHECK_OK(func->AddKernel(base));
   }
+  {
+    // XXX: float16() is not in PrimitiveTypes()
```

Comment:
That's a bummer.

Reply:
We should eventually deal with this somehow. I think we could add a less stable `AllPrimitiveTypes()` that includes `FLOAT16` and eventually mark `PrimitiveTypes()` as deprecated.